### PR TITLE
PIP-249 site redesign

### PR DESCRIPTION
**Pull request by visortelle**
### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

PIP-249: https://github.com/apache/pulsar/issues/19611

Live deployment: https://pulsar-site-pip-249.vercel.app/

Visible changes:
- The Home page.
- Docs page.
- Blog page.
- Ecosystem page.
- Case studies page.
- Resources page.
- Events page.

What’s left:
- [x] Ensure that homepage animation performance is ok, otherwise, replace it with something else. 
- [x] Also, we need to decide what to do with the animation in Firefox. See attached screenshot. Maybe use a static illustration variant for Firefox?
- [x] Announcement bar. The styles are ready. Need to integrate it with docusaurus.config.js (simple). ~0.5h.
- [x] Powered by page. ~0.5h.
- [x] Community page. ~3h.
I’ll resolve these points while @tisonkun reviews the changes.
- [x] Fill users' testimonials with actual content. https://github.com/tealtools/pulsar-site-pip-249/blob/02c9096a0a3da557b7b779bb48bb4aa8f4066ff3/data/testimonials.ts#L7

Needs recheck:
- [x] Contribute page and the recent Client Feature Matrix page.
- [ ] Maybe some hidden pages I don’t know about. I checked all listed under src/pages/*.

### CODE REVIEW

**Comment by asafm** [comment]
image
This is still at work, right?

**Comment by visortelle** [comment]
@asafm please check now.

I think there is a mistake in user testimonials. K.Merkushev is duplicated and J.Joseph is a company.
Maybe you know the right info?

Screenshot 2023-05-14 at 4 47 24 PM

https://github.com/apache/pulsar-site/blob/70dc8e54ccf16308875c4345b28924e79700ce5b/src/pages/index.js#L101

**Comment by visortelle** [comment]
Didn't found this info on LinkedIn, but Google helped.

Screenshot 2023-05-14 at 4 50 52 PM

**Comment by visortelle** [comment]
I noticed issues on mobile. Something goes wrong with background positions when building for production. Will try to figure out the reason.

Local dev:

Screenshot 2023-05-14 at 4 56 18 PM

Production build:

Screenshot 2023-05-14 at 4 56 33 PM

**Comment by visortelle** [comment]
Fixed my previous comment issue.

**Comment by asafm** [comment]
image
The font on the left navigation bar is the same as the one on the main text?

**Comment by asafm** [comment]
image
This section is still work in progress?

**Comment by asafm** [comment]
When surfing on iPhone, the icons for Pulsar features sometime is cut:
image

**Comment by visortelle** [comment]
@tisonkun can you please provide instructions on how to develop abd build ‘/contribute’ and other pages that aren’t available automatically after running ‘npm start’

**Comment by asafm** [comment]
In community, something is not aligned on iPhone 13 mini: 
image

**Comment by asafm** [comment]
The comment up is actually for the entire community page

**Comment by asafm** [comment]
Other than that, it's perfect!

**Comment by visortelle** [comment]
> The font on the left navigation bar is the same as the one on the main text?

Yes, the font-family is the same.

Made it less bold. Probably now it looks better.

Screenshot 2023-05-14 at 6 10 11 PM

**Comment by visortelle** [comment]
@tisonkun maybe there is any way to build a full site in Docker?

I want to be able to build a site in a single command and don't think about tools, or their versions.

Or maybe you have such plans? If yes, I'd be happy to help.

**Comment by tisonkun** [comment]
The file you link is quite old and I don't think it helps here.

https://github.com/apache/pulsar/issues/20272 tracks your need IIUC and let me prepare a patch today against this repo and ping you.

File: docusaurus.config.js
Comment (tisonkun) [inline docusaurus.config.js:400]: It seems we are missing the `contribute` instance here:

```js
        [
            'content-docs',
            /** @type {import('@docusaurus/plugin-content-docs').Options} */
            ({
                id: 'contribute',
                path: 'contribute',
                routeBasePath: 'contribute',
                showLastUpdateAuthor: true,
                showLastUpdateTime: true,
                sidebarPath: require.resolve('./sidebarsDevelopment.js'),
                editUrl: `${githubSiteUrl}/edit/main`,
            }),
        ],
```

Comment (visortelle) [reply docusaurus.config.js:400]: Yes, thank you!
🤦‍♂️

**Comment by visortelle** [comment]
> This section is still work in progress?

@asafm Fixed.

https://pulsar-site-pip-249.vercel.app/contribute/

**Comment by visortelle** [comment]
Will chech 

> Build failed in CI. Perhaps you need to run yarn install and update the lockfile?

Will check in an hour.

File: versioned_docs/version-2.9.x/about.md
Comment (tisonkun) [inline versioned_docs/version-2.9.x/about.md:1]:
```
Compiled with problems:
×
ERROR in ./versioned_docs/version-3.0.x/about.md 1:1096-1152
Module not found: Error: Can't resolve '@site/src/components/BlockLinks' in '/Users/chenzili/Brittani/pulsar-site/versioned_docs/version-3.0.x'
ERROR in ./versioned_docs/version-3.0.x/about.md 1:1152-1206
Module not found: Error: Can't resolve '@site/src/components/BlockLink' in '/Users/chenzili/Brittani/pulsar-site/versioned_docs/version-3.0.x'
```

Need for 3.0 also I think.

File: .github/workflows/ci-precommit.yml
Comment (tisonkun) [inline .github/workflows/ci-precommit.yml:20]: I suggest you revert these commits. You can open a pull request against `main` in your fork to trigger CI also.

Comment (tisonkun) [reply .github/workflows/ci-precommit.yml:20]: Otherwise we cannot run CI upstream.

**Comment by tisonkun** [comment]
Maybe you can tick `Allow edits by maintainers` box so that I can try to fix the build issue for you.

File: .github/workflows/ci-precommit.yml
Comment (visortelle) [reply .github/workflows/ci-precommit.yml]: Reverted :)

**Comment by visortelle** [comment]
2h+. Wow.

Screenshot 2023-05-15 at 6 45 28 PM

@tisonkun I assume, the next builds should take much less time?

There are a few minor tweaks by the designer.
I assume it would be preferable to make new changes in new PRs. 😄

**Comment by tisonkun** [comment]
> I assume, the next builds should take much less time?

This is another tech debt that we maintain too much versions and to build them at once exhaust GitHub Actions Runner memory so we use [`split-version-build.sh`](https://github.com/apache/pulsar-site/blob/22081daf8095676f897a0224351e6116d7d0d183/scripts/split-version-build.sh) to build changed versions one by one. If only a few versions touched, it can be much faster.

Ideally, legacy versions should be archived as a snapshot, which can be another improvement proposal to work on.

**Review by tisonkun — APPROVED** [review]

**Comment by tisonkun** [comment]
LGTM. I think we can merge this patch to prevent further conflicts. Following improvement can be done iteratively.

@asafm @visortelle what do you think? Is there any blocker from merging this as a stable base version?

One thing I can refer to is that @visortelle perhaps you can show this patch and the demo site on dev@pulsar.apache.org so that it doesn't surprise developers that the look of the website is changed.

**Comment by visortelle** [comment]
@tisonkun great 👍

I'll prepare the announcement.

**Comment by Anonymitaet** [comment]
@visortelle thanks for your great contribution. The new look is more clean and professional! 

I have some questions about the doc layout. Can you take a look? Thank you!

### Font and line spacing

The texts look a little dense. What are the default text size and line spacing? Does it make sense to increase them a bit to increase reliability?

https://pulsar-site-pip-249.vercel.app/docs/next/develop-tools/

image

### Font

Do we need to keep consistent on "regular" or "bold"?

https://pulsar-site-pip-249.vercel.app/docs/next/concepts-replication/#replication-patterns

image

### Color

https://pulsar-site-pip-249.vercel.app/docs/next/concepts-throttling/

image

**Comment by Anonymitaet** [comment]
As suggested by @codelipenghui and @mattisonchao, it's more reasonable to update this value to 10ms.

https://pulsar-site-pip-249.vercel.app/
image

Data source:
image

**Comment by Anonymitaet** [comment]
As suggested by @codelipenghui @mattisonchao @BewareMyPower, it's more reasonable to update this value to 10ms.

https://pulsar-site-pip-249.vercel.app/
image

**Comment by visortelle** [comment]
@Anonymitaet thank you for the review. 😀

> The texts look a little dense. What are the default text size and line spacing? Does it make sense to increase them a bit to increase reliability?

The text size is 16px. I increased the line-height a bit from 24px to 26px (1.625).

> Do we need to keep consistent on "regular" or "bold"?

h1 and h2 headers have regular fort by design.

Screenshot 2023-05-19 at 7 36 01 AM

https://www.figma.com/file/0jERoA2DTlfQoj1uH09FxQ/Pulsar-website?type=design&node-id=254-422&t=0DUnBCdzrLQVUWdj-0

For me, it looks normal.
Probably we can keep it for other pages, and make the bold for docs pages. 
@emidio-cardeira WDYT?

> Does it make sense to change "note" background color to more prominent one (e.g. blue) since it requests attention?

Docusaurus has a special syntax for that: https://docusaurus.io/docs/markdown-features/admonitions.

Screenshot 2023-05-19 at 7 54 17 AM

Using `:::info`

Screenshot 2023-05-19 at 7 51 40 AM

Using `:::caution`

Screenshot 2023-05-19 at 7 52 36 AM

For me, it makes sense to distinguish blocks visually that provide more details or give pro tips, and blocks that contain information about potentially destructive actions.
@asafm WDYT?

I found 244 `:::note` entries in the `/docs` dir. Not too many. 
@Anonymitaet if you want, I can help you with changing it to tip/info/caution/danger.

> Tables. Update this light gray to a little darker one?

I updated it to the gray color that is used in other places. For me, contrast is enough. :)

Screenshot 2023-05-19 at 8 05 59 AM

Higher contrast doesn't look good.

Screenshot 2023-05-19 at 8 13 35 AM

@emidio-cardeira WDYT?

> Logo. Why change this from blue to black?

Maybe @emidio-cardeira or @asafm can give a good answer here.

>  it's more reasonable to update this value to 10ms.
Updated.

**Comment by visortelle** [comment]
@Anonymitaet I replaced all `:::note` to `:::info`. Now it's highlighted with blue color.

Screenshot 2023-05-19 at 10 01 17 AM

We can mark some sections with `caution` or `danger` over time.

For example, this block should be marked with `:::caution`:
https://pulsar-site-pip-249.vercel.app/docs/next/administration-zk-bk/#configure-bookkeeper

---

Screenshot 2023-05-19 at 10 05 29 AM

---

Screenshot 2023-05-19 at 10 08 55 AM

---

@tisonkun I splitted it into two commits. Just not sure about versioned_docs. Feel free to revert it if needed:
- [Replace :::note to :::info everywhere except of versioned_docs](https://github.com/apache/pulsar-site/pull/560/commits/b190bf687e17526cd69640212fce7484d90b83f8)
- [Replace :::note to :::info in versioned_docs](https://github.com/apache/pulsar-site/pull/560/commits/43b4d8afcc4f9ee49022ccd1790fd90aa7cd2e41)

**Comment by emidio-cardeira** [comment]
@visortelle 

**> Logo. Why change this from blue to black?**

We increased a lot the number of colors on the website. So we're trying to make Pulsar visual language more friendlier and approachable.
And because we have all of those colors we think the logo should be more neutral and that way to increase logo visibility.

**Alert banner**
I'm ok in using a light grey like this one, if we need to keep the option.
Design spec here https://www.figma.com/file/0jERoA2DTlfQoj1uH09FxQ/Pulsar-website?type=design&node-id=1165-1046&t=thPf8NEKpouB85vO-4

btw can we remove the shadow please :)



**Body**
This font is a bit smaller that is true
Let's increase the Body P-M style to 
size 18pt
letter spacing -6%
line height 28 

WDYT?

**Titles**
Let's create a new H with 
Screenshot 2023-05-19 at 17 51 51

**Comment by visortelle** [comment]
@emidio-cardeira 
✅ alert shadow is removed
✅ text font size is increased to 18px
✅ h2 changed to semibold on the docs page

**Comment by visortelle** [comment]
@asafm @tisonkun @emidio-cardeira @Anonymitaet

If anyone has anything to add, please do it. 😄
Otherwise, let's merge it. 🚀

**Comment by asafm** [comment]
I'm back from vacation. I'll take a look.
Did you end up posting the announcement in the DEV mailing list @visortelle ?

**Comment by asafm** [comment]
Looks good to me.
We can plan the next steps after we merge this one.

**Comment by visortelle** [comment]
> Did you end up posting the announcement in the DEV mailing list @visortelle ?

@asafm Yes, here it is:
https://lists.apache.org/thread/sjlkpt1zb29k3vfbqvgv271brkf0tcsr

**Comment by Anonymitaet** [comment]
Hi @visortelle thanks for your quick updates!

In technical writing, different admonitions have different meanings, they are not exchangeable. 

For example,

> DANGER—Indicates a hazardous situation that, if not avoided, will result in serious problems
> This text is to be limited to the most extreme situations. Call out the hazard and steps to
> avoid it.
> 
> CAUTION—Indicates a hazardous situation that, if not avoided, could result in minor or moderate
> problems. Call out the hazard or consequence and steps to avoid it.
> 
> NOTE—Contains additional information to emphasize or supplement important points of the main
> text. Information in a note should not add value to the preceding text; if it does, use an important
> alert instead.
> 
> TIP—Provides helpful hints for completing a task. Do not use a tip to give essential information. A
> tip might provide an alternative method for completing the task that precedes it.

So can we customize the note color to blue (and not update note to other admonitions)?

**Comment by visortelle** [comment]
@Anonymitaet 
I think it depends on the style-guide you prefer to use. :)
I changed it to blue as you wish.

Screenshot 2023-05-22 at 9 45 23 AM

**Comment by Anonymitaet** [comment]
@visortelle that's great! 
I've triggered the tests to run, let's wait for the pass.

**Comment by tisonkun** [comment]
Merging...

Thank you!

**Comment by visortelle** [comment]
@tisonkun @Anonymitaet at https://pulsar.apache.org I still see the old version.
Maybe we should trigger some CI job to deploy the merged version?

**Comment by tisonkun** [comment]
I trigger it at https://github.com/apache/pulsar-site/actions/runs/5053822355/jobs/9068057937

Weird the push commit don't trigger it :/

**Comment by visortelle** [comment]
@tisonkun the reason was that one of my commits contained the `[ci skip]` instruction. 
The squashed commit message is a concatenation of all commit messages, therefore all CI jobs were skipped.

Screenshot 2023-05-23 at 8 26 52 AM

**Comment by tisonkun** [comment]
I don't cleaning the commit message in this case :D

**Comment by Anonymitaet** [comment]
Hi all, 

More developers are checking the new website today, and they think it's awesome!

At the same time, I get feedback from @mattisonchao and @codelipenghui, shown in the image below

image

**Comment by visortelle** [comment]
Hi @Anonymitaet,

> More developers are checking the new website today, and they think it's awesome!

Glad to hear that. :)

Regarding font size, let's ask @emidio-cardeira.

Screenshots to simplify the comparison:

**2rem (current)**

Screenshot 2023-05-24 at 8 09 52 AM

**1.5rem**

Screenshot 2023-05-24 at 8 10 03 AM

**1.5rem + less width**

Screenshot 2023-05-24 at 8 13 13 AM

**Comment by emidio-cardeira** [comment]
@visortelle @Anonymitaet 
Maybe let's wait for more feedback before making any changes.
let the dust settle a bit and then evaluate

**Pull request merged**
